### PR TITLE
I18n localization.

### PR DESCRIPTION
**Pull request by lexjacobs**
To review:

### Please update your local npm modules by typing the following in a terminal:

* `cd your/path/to/goldstone-server/`
* `npm install`

#### After getting the containers and server up and running (unless you're in the new docker-celery environment in which case you just have to get the container running) . . .

Type `grunt watch`

This will allow you to make experimental changes to .po files and have the results render on-screen via a full page refresh.

### Things you might try:

* navigate to `http://localhost:8000/#settings` and change the language file to `Chef`.
* Browse around the site and see if you can find any untranslated strings remaining.
* Create your own .po file and drop it into `goldstone-server/goldstone/static/i18n/po_files/`
* Do a full page refresh and navigate to the settings page to see your new language populate the settings language drop-down.
* Install OpenTrail and make sure everything is translated into Chef, or your own .po language.
* Check out the I18N.MD file and see that it makes sense.

chef

_And now, please excuse me while I find some replacement eyeballs..._

### CODE REVIEW

File: docs/I18N.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+* <a href="#implement">How to implement i18n client-side so that the text strings of your choice are translated to the user-selected language.</a>
+* <a href="#base-html">Special procedures related to having the client translate the django-served base.html file.</a>
+
+### _This document assumes you have installed Goldstone according to the HACKING.md document, and have node/npm installed._
+
+Make sure your dev environment is up to date with the latest npm dependencies with:
+
+`$ npm install`
+
+Please keep in mind that there is no direct relationship between .po files (which are standard for gettext), and JavaScript. So in order for changes or additions to the .po files in `goldstone/static/i18n/po_files` to effect the client rendering, you have to make sure that modifications are made while the Grunt.js task runner is operating by running `grunt watch` in a terminal tab. Alternately you could run `grunt po2json` from the terminal after making changes to a .po file.
```
Comment (stugots) [inline docs/I18N.md]: "...to AFFECT the client rendering..."

Comment (stugots) [inline docs/I18N.md]: "So in order for changes or additions to the .po files in `.......` to..." ->   "For changes in .po files in `.......` to..."

File: docs/I18N.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+<a name="about"></a>
+## How i18n is implemented for the client in Goldstone.
```
Comment (stugots) [inline docs/I18N.md]: How i18n is implemented in the Goldstone client

File: docs/I18N.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+The Goldstone i18n scheme is based on the gettext standard, with is a very common method of internationalizing an app, and with which many translators are already familiar. There are multiple tools and libraries available so that the same .po files can be applied to client and server environments. Goldstone uses a popular library, **[Jed.js](https://github.com/SlexAxton/Jed)**, to make available the gettext functions. And to make the .po files readable by Jed.js, we chose the recommended adjunct library, po2json which makes for a Jed.js-readable json object.
```
Comment (stugots) [inline docs/I18N.md]: with is a very  --> which is a very

Comment (stugots) [inline docs/I18N.md]: There are multiple tools and libraries available so that ==> Tools and libraries are available, so that

File: docs/I18N.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+With the `grunt watch` environment running, any additions of new .po files, or modification of existing .po files will trigger a Grunt task called **po2json** that will create a new json object that will be served by django, and made available to the JavaScript client.
```
Comment (stugots) [inline docs/I18N.md]: Capitalize Django

File: docs/I18N.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+With each browser page refresh, the client initialization process includes parsing the translated i18n json object, and making it readable by the Jed.js library that adapts the standard gettext functions for JavaScript.
+
+<a name="import"></a>
+## How to import a new .po file into the front-end i18n scheme so it's available as a selection in the client.
+
+When adding a new .po file, keep in mind that the actual name of the file prior to the `.po` (eg: _Bocce_ from **Bocce.po**) is what will be used to dynamically populate the choice of languages that can be selected in the client via the settings page. Capitalization will be carried over, so be sure to name your .po file exactly as you'd like it applied as a language choice.
+
+Adding a file to or changing a file in `goldstone/static/i18n/po_files` (with the `grunt watch` task running) will trigger a re-rendering of `goldstone/static/i18n/po_json/i18n_combined.json`. This file is used to by the Jed.js library as described in the section above.
```
Comment (stugots) [inline docs/I18N.md]: This file is used to by the ==> This file is used by the

**Comment by stugots** [comment]
:+1: after my comments are considered.

**Comment by lexjacobs** [comment]
All suggested edits thus far implemented.

**Comment by jxstanford** [comment]
Getting lots of these when I run grunt.  Server is running...

```Chrome 45.0.2454 (Mac OS X 10.10.5) zoomablePartitionview.js spec "before each" hook FAILED
	TypeError: goldstone.translate is not a function```

**Comment by lexjacobs** [comment]
@jxstanford , please `git up` for latest changes that address your documented karma test issues. Please confirm tests are passing on your local machine. Thank you.

**Comment by jxstanford** [comment]
:+1:

**Comment by lexjacobs** [comment]
Great thanks. I have just a clean up commit to make (remove fake .po file) and then will merge.

**Pull request merged**